pipeline_dp/combiners.py

```diff
@@ -217,6 +217,28 @@ def metrics_names(self) -> List[str]:
         return self._metrics_to_compute
 
 
+# Cache for namedtuple types. It is should be used only in
+# '_get_or_create_named_tuple()' function.
+_named_tuple_cache = {}
+
+
+def _get_or_create_named_tuple(type_name: str,
+                               field_names: tuple) -> 'MetricsTuple':
+    """Creates namedtuple type with a custom serializer."""
+
+    # The custom serializer is required for supporting searialization of
+    # namedtuples in Apache Beam.
+    cache_key = (type_name, field_names)
+    named_tuple = _named_tuple_cache.get(cache_key)
+    if named_tuple is None:
+        named_tuple = collections.namedtuple(type_name, field_names)
+        named_tuple.__reduce__ = lambda self: (_get_or_create_named_tuple,
```

Comment:
Setting `__reduce__` defines custom serializer. 
Custom serialzier should return a tuple (`func_to_create_object_type`, arguments for this function)